components/ai_chat/resources/page/components/main/index.tsx

```diff
@@ -185,12 +214,197 @@ function Main() {
+      document.removeEventListener('dragleave', handleDragLeave)
+      document.removeEventListener('dragover', handleDragOver)
+      document.removeEventListener('dragend', handleDragEnd)
+      document.removeEventListener('mouseleave', handleMouseLeave)
+      document.removeEventListener('focusout', handleDocumentFocusOut)
+      window.removeEventListener('blur', handleWindowBlur)
+      document.removeEventListener('visibilitychange', handleVisibilityChange)
+    }
+  }, [])
+
+  // Listen for iframe drag notifications via mojom
+  React.useEffect(() => {
+    const api = getAPI()
+
+    const dragStartId = api.conversationEntriesFrameObserver.dragStart
+      .addListener(() => {
+        setIsDragActive(true)
+        setIsDragOver(true)
+      })
+
+    const dragEndId = api.conversationEntriesFrameObserver.dragEnd
+      .addListener(() => {
+        setIsDragActive(false)
+        setIsDragOver(false)
+      })
+
+    return () => {
+      api.conversationEntriesFrameObserver.removeListener(dragStartId)
+      api.conversationEntriesFrameObserver.removeListener(dragEndId)
+    }
+  }, [])
+
+  const handleOverlayDrop = async (e: React.DragEvent) => {
+    e.preventDefault()
+    e.stopPropagation()
+    setIsDragActive(false)
+    setIsDragOver(false)
+
+    const files = Array.from(e.dataTransfer?.files || []).filter(isImageFile)
+
+    if (files.length === 0) {
+      return
+    }
+
+    try {
+      const uploadedFiles = await Promise.all(
+        files.map(file => convertFileToUploadedFile(file))
+      )
+      conversationContext.processDroppedImages(uploadedFiles)
+    } catch (error) {
+      // Silently fail - error will be handled by the upload system
+    }
+  }
+
+  const handleOverlayDragOver = (e: React.DragEvent) => {
+    e.preventDefault()
+  }
+
   return (
-    <main className={classnames({
+    <main
+      className={classnames({
       [styles.main]: true,
       [styles.mainPanel]: !aiChatContext.isStandalone,
-      [styles.mainMobile]: aiChatContext.isMobile
-    })} ref={setMainElement}>
+      [styles.mainMobile]: aiChatContext.isMobile,
+      [styles.dragOver]: isDragOver
+    })}
+      ref={setMainElement}
+    >
+      {/* Transparent overlay that captures drag events over iframe */}
+      {isDragActive && (
+        <div
+          className={styles.dragDetectionOverlay}
+          onDragOver={handleOverlayDragOver}
+          onDrop={handleOverlayDrop}
+        />
+      )}
+
+      {isDragOver && (
+        <div className={styles.dragOverlay}>
+          <div className={styles.dragOverlayContent}>
+            <div className={styles.dragIcon}></div>
+            <div className={styles.dragTitle}>
+              {getLocale(S.CHAT_UI_DROP_IMAGES_HERE_LABEL)}
+            </div>
+            <div className={styles.dragDescription}>
+              {getLocale(S.CHAT_UI_DROP_FILES_DESCRIPTION)}
+            </div>
+          </div>
+        </div>
+      )}
```

Comment:
let's move this into it's own component (and the associated CSS).

Reply:
Fixed in a5543fb5bfa40f4cead2aa7c7cbc0b0ea0293517